torch/csrc/distributed/c10d/SymmetricMemory.cpp

```diff
@@ -203,3 +203,49 @@ TORCH_API bool has_multicast_support(
   return allocator->has_multicast_support(device_idx);
 }
 } // namespace c10d::symmetric_memory
+
+namespace {
+
+at::Tensor one_shot_all_reduce_meta(
+    const at::Tensor& input,
+    std::string reduce_op,
+    std::string group_name) {
+  return at::empty_like(input);
+}
+
+TORCH_LIBRARY_FRAGMENT(symm_mem, m) {
+  m.def(
+      "multimem_all_reduce_(Tensor(a!) input, str reduce_op, str group_name) -> Tensor(a!)");
+  m.def(
+      "multimem_one_shot_all_reduce(Tensor input, str reduce_op, str group_name) -> Tensor");
+  m.def(
+      "multimem_one_shot_all_reduce_out(Tensor input, str reduce_op, str group_name, Tensor(a!) out) -> Tensor(a!)");
+  m.def(
```

Comment:
This might be pt2 compliant because there's a meta definition. (see https://docs.google.com/document/d/1_W62p8WJOQQUzPsJYa7s701JXt0qf2OfLub2sbkHOaU/edit?tab=t.0 to test - try `opcheck`).